### ISSUE TITLE
The shape of LazyArray

### ISSUE THREAD

**Issue by yangguohao** [jcmgray/autoray#14]
```
from autoray.lazy import Variable
from autoray import do

a = Variable(shape=(4,))
b = do('random.normal', size=(4,), like='numpy')
c = do('asarray', b, like='torch')

index = [[2, 3]]
print(a[index].shape)
print(b[index].shape)
print(c[index].shape)
print('----------------------------')
index = [[[0, 1, 2, 3]]]
print(a[index].shape)
print(b[index].shape)
print(c[index].shape)

# (1,)
# (1, 2)
# torch.Size([2])
# ----------------------------
# (1,)
# (1, 1, 4)
# torch.Size([1, 4])
```

The shape of `LazyArray` are different from the `torch.Tensor` and `numpy.array` after the getitem operation in these two situation.

Here are my rough thoughts:
1. Let the `LazyArray` shape consistent with the `numpy.array`.
2. Let the `torch.Tensor` shape consistent with the `numpy.array`(may use the wrapper for getitem when dealing with the `torch.Tensor`)

**Comment by jcmgray** [comment]
I think matching `numpy` is the desired behavior. There are so many fancy indexing options and I seem to have missed this one in the lazy getitem (takes a while to work out what is intended here!). I also didn't know that torch differed on some indexing, or indeed what the logic is that gives e.g. a shape of (1, 4)...

Is there a functional form a la `np.take`? Supporting that would be the cleanest way to have uniform behavior.

**Comment by yangguohao** [comment]
Yeah, it's a good option to match the `lazy` with `numpy`, and I'm not sure what does "a la `np.take`" mean, that means a `take` function for LazyArray? I have tried using `np.take` or `torch.take` and seems like they have uniform behavior. I can open a PR add `take` function for LazyArray

**Comment by jcmgray** [comment]
> and I'm not sure what does "a la np.take" mean, that means a take function for LazyArray?

Sorry I just meant, I know `numpy.take(array, indices)` is the functional form of some types of fancy indexing `array[indices]`, but don't know if it includes this use case, or there is another function.

**Comment by yangguohao** [comment]
I see, the `np.take`, `torch.take`, `tensorflow.experimental.numpy.take` did include this use case and have uniform behavior

**Comment by yangguohao** [comment]
I see, the `np.take`, `torch.take`, `tensorflow.experimental.numpy.take` did include this use case and have uniform behavior

**Comment by jcmgray** [comment]
Nice. So useful things to do here would probably be to both:

- [ ] support fancy indexing in `autoray.lazy.getitem`
- [ ] implement `autoray.lazy.take`

Let me know if you fancy implementing either of these.

**Comment by yangguohao** [comment]
> Nice. So useful things to do here would probably be to both:
> 
> * [ ]  support fancy indexing in `autoray.lazy.getitem`
> * [ ]  implement `autoray.lazy.take`
> 
> Let me know if you fancy implementing either of these.

Hi, @jcmgray I open this new PR[(https://github.com/jcmgray/autoray/pull/15)] for this

**Comment by jcmgray** [comment]
Closed by #15.
